k2/csrc/array_ops_inl.h

```diff
@@ -477,6 +477,27 @@ Array2<T> ToContiguous(const Array2<T> &src) {
   return ans;
 }
 
+template <typename T>
+bool Equal(const Array1<T> &a, const Array1<T> &b) {
+  if (a.Dim() != b.Dim()) {
+    K2_LOG(FATAL) << "Dimension mismatch: " << a.Dim() << " vs. "
+                  << b.Dim();
+  }
+  ContextPtr c = GetContext(a, b);
+  const T *a_data = a.Data(), *b_data = b.Data();
+  if (c->GetDeviceType() == kCpu) {
+    return memcmp((void*)a_data, (void*)b_data, sizeof(T) * a.Dim()) == 0;
+  } else {
+    Array1<int32_t> is_same(c, 1, 1);
```

Comment:
Should we add an additional check here?

```cpp
K2_CHECK_EQ(c->GetDeviceType(), kCuda);
```

Reply:
that may be checked inside eval(), I think I am not going to put these checks in places like this.